ddspipe_core/include/ddspipe_core/types/dynamic_types/types.hpp

```diff
@@ -35,7 +38,11 @@ struct DynamicTypeData : public core::IRoutingData
     DDSPIPE_CORE_DllAPI
     virtual types::TopicInternalTypeDiscriminator internal_type_discriminator() const noexcept override;
 
-    fastrtps::types::DynamicType_ptr dynamic_type{nullptr};
+    fastdds::dds::DynamicType::_ref_type dynamic_type{nullptr};
+
+    fastcdr::string_255 type_name{};
+
+    fastdds::dds::xtypes::TypeIdentifier type_id{};
```

Comment:
This scenario will be addressed regardless, as the minimal `TypeIdentifier` will be transmitted and then set as TypeIdentifier1 when creating the `TypeIdentifiersPair` (We have already observed that the order does not impact the retrieval of the `TypeObject`). Nonetheless, I will change it to `TypeIdentifiersPair` so that, if we decide to transmit both complete and minimal identifiers in the future, we will need some minimal adjustments.

Reply:
Update:
- It seems like we are only keeping one identifier after all, and it's always the complete one (in the DynTypesParticipant case). So I'd say we should add that as comments of the attributes.
- Regarding the `type_name`, apparently `get_name` only works for the complete case, so I guess we should keep it if this type is meant for both complete and minimal cases (although my vote is for having only the complete one).